components/yaml/introduction.rst

```diff
@@ -250,6 +236,64 @@ representation of the object.
     parsers will likely not recognize the ``php/object`` tag and non-PHP
     implementations certainly won't - use with discretion!
 
+.. _invalid-types-and-object-serialization:
+
+Handling Invalid Types
+~~~~~~~~~~~~~~~~~~~~~~
+
+By default the parser will encode invalid types as ``null``. You can make the
+parser throw exceptions by using the ``PARSE_EXCEPTION_ON_INVALID_TYPE``
+flag::
+
+    $yaml = '!php/object:O:8:"stdClass":1:{s:5:"foo";s:7:"bar";}';
+    Yaml::parse($yaml, Yaml::PARSE_EXCEPTION_ON_INVALID_TYPE); // throws an exception
+
+Similarly you can use ``DUMP_EXCEPTION_ON_INVALID_TYPE`` when dumping::
+
+    $data = new \stdClass(); // by default objects are invalid.
+    Yaml::parse($data, Yaml::DUMP_EXCEPTION_ON_INVALID_TYPE); // throws an exception
+
+.. _objects-for-mappings:
+
+    echo $yaml; // { foo: bar }
+
+Date Handling
+~~~~~~~~~~~~~
+
+By default the YAML parser will convert unquoted strings which look like a
+date or a date-time into a Unix timestamp; for example ``2016-05-27`` or
+``2016-05-27T02:59:43.1Z`` (ISO-8601_)::
+
+    Yaml::parse('2016-05-27'); // 1464307200
+
+You can make it convert to a ``DateTime`` instance by using the ``PARSE_DATETIME``
+flag::
+
+    $date = Yaml::parse('2016-05-27', Yaml::PARSE_DATETIME);
+    var_dump(get_class($date)); // DateTime
+
+Dumping Multi-line Literal Blocks
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In YAML multiple lines can be represented as literal blocks, by default the
+dumper will encode multiple lines as an inline string::
+
+    $string = ["string" => "Multiple\nLine\nString"];
+    $yaml = Yaml::dump($string);
+    echo $yaml; // string: "Multiple\nLine\nString"
+
+You can make it use a literal block with the ``DUMP_MULTI_LINE_LITERAL_BLOCK``
+flag::
+
+    $string = ["string" => "Multiple\nLine\nString"];
+    $yaml = Yaml::dump($string, 2, 4. Yaml::DUMP_MULTI_LINE_LITERAL_BLOCK);
```

Comment:
Updated

On Mon, Jun 27, 2016 at 08:03:05AM -0700, Jules Pietri wrote:
>    In [1]components/yaml/introduction.rst:
> 
>  > +
>  > +Dumping Multi-line Literal Blocks
>  > +~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>  > +
>  > +In YAML multiple lines can be represented as literal blocks, by default the
>  > +dumper will encode multiple lines as an inline string::
>  > +
>  > +    $string = ["string" => "Multiple\nLine\nString"];
>  > +    $yaml = Yaml::dump($string);
>  > +    echo $yaml; // string: "Multiple\nLine\nString"
>  > +
>  > +You can make it use a literal block with the ``DUMP_MULTI_LINE_LITERAL_BLOCK``
>  > +flag::
>  > +
>  > +    $string = ["string" => "Multiple\nLine\nString"];
>  > +    $yaml = Yaml::dump($string, 2, 4. Yaml::DUMP_MULTI_LINE_LITERAL_BLOCK);
> 
>    should be a coma after 4
> 
>    —
>    You are receiving this because you were mentioned.
>    Reply to this email directly, [2]view it on GitHub, or [3]mute the thread.
> 
>    Reverse link: [4]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/symfony/symfony-docs/pull/6622#discussion_r68592452
>    2. https://github.com/symfony/symfony-docs/pull/6622/files/9cc982fe591230b1b3a9be380f9859727829711b#r68592452
>    3. https://github.com/notifications/unsubscribe/AAgZcZOz_Sp0D7xXpyI7Fbf4a8vzhxSMks5qP-YpgaJpZM4IoXrx
>    4. https://github.com/symfony/symfony-docs/pull/6622/files/9cc982fe591230b1b3a9be380f9859727829711b#r68592452